src/components/layout.js

```diff
@@ -116,12 +126,12 @@ const Sidebar = props => {
         }}
       >
         {props.children}
-        <Reactions />
+        {
+          showPaywall === false
+            && <Reactions />
+        }
         {showPaywall ? (
-          <>
-            <Paywall floating />
-            <Paywall />
-          </>
+          <Paywall page={props.location.pathname} />
```

Comment:
Why is it useful for the paywall to get the page name?

Reply:
To get the changes when the user changes the page. According to the way react update the children the `lock content function` wasn't fired if the user just change the page